client/src/extension.ts

```diff
@@ -37,27 +37,25 @@ export function activate(context: vscode.ExtensionContext) {
         };
     } else {
         console.log('Starting without debug');
-        // Name of the launcher class which contains the main.
-        const main: string = 'org.drools.lsp.server.DroolsLspLauncher';
 
-        const {JAVA_HOME} = process.env;
+        const javaHome = getJavaHome();
 
-        // If java home is available continue.
-        if (JAVA_HOME) {
-            // If java home is available continue.
-            console.log(`Using java from JAVA_HOME: ${JAVA_HOME}`);
-            let excecutable: string = path.join(JAVA_HOME, 'bin', 'java');
+        let executable: string = `java`;
 
-            // path to the launcher.jar
-            let classPath = path.join(__dirname, '..', '..', 'drools-lsp-server', 'target', 'drools-lsp-server-jar-with-dependencies.jar');
-            const args: string[] = ['-cp', classPath];
-
-            serverOptions = {
-                command: excecutable,
-                args: [...args, main],
-                options: {}
-            };
+        if (javaHome) {
+            // If java home is available, compose a path
+            executable = path.join(javaHome, 'bin', 'java');
         }
+
+        // path to the launcher.jar
+        let serverJar = path.join(__dirname, '..', '..', 'drools-lsp-server', 'target', 'drools-lsp-server-jar-with-dependencies.jar');
```

Comment:
Is it possible to pack a standalone version of `drools-lsp-server-jar-with-dependencies.jar` inside the vsix?
Otherwise, this is assuming that this jar is on a specific path that could not always be the case for users.

We have some cases where we pack a Quarkus jar app inside the vsix and run it inside the extension. Looks the same situation.

Also, I think it is worth checking if `serverJar` file exists.

Reply:
Thank you for the suggestion, @caponetto 

I'll pack the jar inside the vsix. I'll work on that in a separated JIRA https://issues.redhat.com/browse/DROOLS-7342

> Also, I think it is worth checking if `serverJar` file exists.

Thanks. I added the check.